toolkit/tools/imagegen/configuration/networkconfig.go

```diff
@@ -0,0 +1,309 @@
+}
+
+func (n *Network) validateIPAddress(ip string) (err error) {
+	if ip == "" && (n.BootProto == "" || n.BootProto == "dhcp") {
+		return
+	} else if net.ParseIP(ip) == nil {
+		return fmt.Errorf("invalid ip address (%s)", ip)
+	}
+
+	return
+}
+
+// ipAddrIsValid returns an error if ip, gateway, netmask or nameserver inputs are invalid ip addresses
+func (n *Network) ipAddrIsValid() (err error) {
+	if err = n.validateIPAddress(n.Ip); err != nil {
+		return fmt.Errorf("invalid input for ip: %w", err)
+	}
+
+	if err = n.validateIPAddress(n.NetMask); err != nil {
+		return fmt.Errorf("invalid input for netmask: %w", err)
+	}
+
+	if err = n.validateIPAddress(n.GateWay); err != nil {
+		return fmt.Errorf("invalid input for gateway: %w", err)
+	}
+
+	for _, nameserver := range n.NameServers {
+		if err = n.validateIPAddress(nameserver); err != nil {
+			return fmt.Errorf("invalid input for nameserver: %w", err)
+		}
+	}
+
+	return
+}
+
+// deviceIsValid returns an error if the Device name is empty
+func (n *Network) deviceIsValid() (err error) {
+	n.Device = strings.TrimSpace(n.Device)
+	if n.Device == "" {
+		return fmt.Errorf("invalid input for device, device cannot be empty")
+	}
+	return
+}
+
+func findBootIfValue() (deviceAddr string, err error) {
+	const macAddressStartIndex = 3
+
+	bootifValue, ferr := GetKernelCmdLineValue("BOOTIF")
+	bootifValue = strings.TrimSpace(bootifValue)
+	if ferr != nil || bootifValue == "" {
+		err = ferr
+		return
+	}
+
+	// The bootif value in the cmdline set by pxelinux is of the following format:
+	// bootif=01-MAC Address, where each byte value of the MAC address is separated
+	// by dashes instead of colons. Therefore, we're reading from the 10th spot of the
+	// string to obtain the MAC address and then replace the dashes with colons
+	deviceAddr = strings.Replace(bootifValue[macAddressStartIndex:len(bootifValue)], "-", ":", -1)
+	return
+}
+
+func checkNetworkDeviceAvailability(networkData Network) (deviceName string, err error) {
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return
+	}
+
+	if networkData.Device == "bootif" {
+		networkData.Device, err = findBootIfValue()
+		if err != nil {
+			logger.Log.Errorf("Failed to read bootif value from /proc/cmdline")
+			return
+		}
+	}
+
+	for _, iface := range ifaces {
+		if networkData.Device == iface.Name {
+			deviceName = networkData.Device
+			return
+		} else {
+			ifaceAddr := strings.TrimSpace(iface.HardwareAddr.String())
+
+			if strings.Contains(networkData.Device, ":") {
+				if ifaceAddr != "" && strings.EqualFold(ifaceAddr, networkData.Device) {
+					deviceName = iface.Name
+					return
+				}
+			}
+		}
+	}
+
+	return
+}
+
+func populateMatchSection(networkData Network, fileName, deviceName string) (err error) {
+	matchSection := fmt.Sprintf("[Match]\nName=%s\n", deviceName)
+	err = file.Append(matchSection+"\n", fileName)
```

Comment:
Why the additional new line here and not in `populateNetworkSection`? Also, why not add it to the rest of the string above?

Reply:
That's because I want to have a line gap between the [Match] and the [Network] section for better reading when looking at this file.

Reply:
Got it. In that case, I'd move adding that additional line out of this function - the function should not have to know in which order it is used, that should be more to the caller and the function itself should only worry about filling the section it's meant to fill.

Reply:
Makes sense. Fixed.